assets/src/components/Sidebar/CampaignSidebar.js

```diff
@@ -85,19 +86,20 @@ export class CampaignSidebar extends Component {
       meta: null,
       parent: null,
     };
-    this.handleThemeChange = this.handleThemeChange.bind( this );
   }
 
   componentDidMount() {
     wp.data.subscribe( () => {
       const meta = wp.data.select( 'core/editor' ).getEditedPostAttribute( 'meta' );
+
       if ( meta ) {
         let theme = meta[ 'theme' ];
         if ( theme === '' ) {
           theme = 'default';
         }
         if ( !isShallowEqual( this.state.meta, meta ) ) {
           this.setState( { meta: meta } );
+          savePreviewMeta();
```

Comment:
Be aware that `CampaignSidebar` will only mount if the icon is clicked! I think we might also need to update this preview if the sidebar isn't opened.

Reply:
Hmmm good point, so maybe we need to call it on `DOMContentLoaded` after all. Or maybe there is a better editor hook for that. I'll dig into it.

Reply:
I just checked: looks like the component is mounted even if the Sidebar is not displayed, so `savePreviewMeta` is called on load.

Reply:
This is not relevant anymore.